src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,66 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { Filters } from '../../types/Filters';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+const filters = ['All', 'Active', 'Completed'];
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+type Props = {
+  setSelectedFilter: React.Dispatch<React.SetStateAction<Filters>>;
+  setQuery: React.Dispatch<React.SetStateAction<string>>;
+  query: string;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  setSelectedFilter,
+  setQuery,
+  query,
+}) => {
+  const handleSelectFilter = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedFilter(event.target.value as Filters);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value.trimStart());
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select data-cy="statusSelect" onChange={handleSelectFilter}>
+            {filters.map(filter => (
+              <option value={filter.toLowerCase()} key={filter}>
```

Comment:
You are generating keys for your list items on render. It's not the best approach because React uses keys to identify which items have changed, are added, or are removed and it may negatively impact performance. Keys should be stable, predictable, and unique for every set of elements. You can read more about it [here](https://medium.com/blackrock-engineering/5-common-mistakes-with-keys-in-react-b86e82020052).